content/english/java-basics/activity-6.md

```diff
@@ -3,4 +3,182 @@ title: "Object and Classes"
+
+Let's learn about the different parts in a class below:
+
+{{% notice note %}}
+#### Class
+A class is a blueprint or prototype of a new type of object. In general, a class contains three important parts: (1) data fields/instance variables (2) constructor (3) methods. 
+
+<img src="../images/class.png" height="400"/> 
+
+1. <b>data fields/instance variables</b>: These are the variables an object of this class has access to.
+                                           For example: A `Car` class may have the field `tire` of class `Tire`.
+
+2. <b>constructor</b>: This is a method we call to create an object of this Class.
+
+3. <b>methods</b>: These are methods an object of this class will be able to call to perform certain tasks.
+
+For example:
+```java
+public class SampleClass{
+    // (1) data fields/instance variables
+    private String name; // example
+
+    // (2) constructor
+    public SampleClass(){
+        name = "example";
+    }
+    // (3) methods
+    public sampleMethod(){}
+}
+```
+{{% /notice %}}
+
+### 2. Try it Out 🐥!
+Let's make a `Bird` class to represent Patrick 🐥 and all of his Bird friends by following the steps below!
+
+<iframe height="600px" width="100%" src="https://repl.it/@nuevofoundation/JavaBasicsBird?lite=true#Bird.java" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>
+
+#### First, we start by defining the class name in the format `public` `class` `name`.
+```java
+public class Bird{}
+```
+<br />
+
+####  Second, let's declare all the fields of the Bird class: species, name, hobby, age, loveMusic.
+Each fields is declared in the format: `private` `data type` `name` `;`.
+1. We declare all 5 fields as `private` in the class `Bird`. This ensures these data can only be accessed within this class.
+
+2. Let's identify the appropriate type for each field:
+- `species` should be a `String` that store for exmaple: "duck", "swan", "owl".
+- `name` should be a `String` that store for example: "Patrick".
+- `hobby` should be a `String` that store for example: "play basketball".
+- `age` should be an `int` that store for example: 25.
+- `loveMusic` should be a `boolean` that store either true or false.
+
+For example, to declare `species` as a private field of class `Bird`, you would put `private String species;` .
+
+3. Finish creating the other 4 fields of the Bird class!
+
+<br />
+
+#### Third, let's create the constructor for the class `Bird`. 
+Usually, constructor is the method that initialializes value to all the fields in a class. It has the format `public` `class name` `(parameter)`. Since we have 5 fields in this class, the constructor will take in 5 parameter/inputs.
+```java
+public Bird (String species_input, String name_input, String hobby_input, int age_input, boolean loveMusic_input){
```

Comment:
Is the convention in Java to use snake_case for parameters or camelCase? I thought it was camelCase.